include/odp/api/spec/pool.h

```diff
@@ -59,6 +59,9 @@ odp_pool_t odp_pool_create(const char *name, const odp_pool_param_t *param);
 /**
  * Destroy a pool previously created by odp_pool_create()
  *
+ * The pool to be destroyed must not be in use in any CoS. If it is, assign
+ * another pool to the CoS, or destroy the CoS, before calling this function.
+ *
```

Comment:
These instructions are a bit specific to classifier API - to be located here in the pool API. I think it should enough to move the "note" section from below up here (remove "@note" tag while doing that), and edit it to say that the pool must not be in-use (including packet input/output, classifier, timer, ...) when calling this function.